include/fc/thread/future.hpp

```diff
@@ -160,10 +184,13 @@ namespace fc {
 
       template<typename CompletionHandler>
       void on_complete( CompletionHandler&& c ) {
-        _on_complete( new detail::completion_handler_impl<CompletionHandler,void>(fc::forward<CompletionHandler>(c)) );
+        _on_complete( new detail::completion_handler_impl<CompletionHandler,void>(std::forward<CompletionHandler>(c)) );
       }
     protected:
-      ~promise(){}
+      promise( const char* desc ):promise_base(desc){}
+      promise( bool fulfilled, const char* desc ){
+          if( fulfilled ) set_value();
+      }
```

Comment:
The description doesn't really matter if the promise is already fulfilled, because we'll never wait on it, so no one will ever want to know the description of what we're waiting on... But then, why take a description at all? Can we not just omit this argument entirely if the promise is constructed already-fulfilled, like we do with `promise<NotVoid>`?

Reply:
Note that the original code had
```
promise( bool fulfilled, const char* desc FC_TASK_NAME_DEFAULT_ARG ){
```

Now promises are always constructed via `promise<T>::create`. The default description has move to the declaration of that:
```
static ptr create( bool fulfilled, const char* desc FC_TASK_NAME_DEFAULT_ARG )
      {
         return ptr( new promise<void>( fulfilled, desc ) );
```

...which means that effectively you can create a fulfilled promise without *specifying* a description (although under the hood it will use a default pointer).